src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1608,9 +1681,13 @@ protected override void InvokeDefaultAction(string path)
             DirectoryInfo directory,
             bool recurse,
             uint depth,
+            InodeTracker tracker,   // This will be non-null only if the user invoked the -FollowSymLinks switch parameter.
             bool nameOnly,
             ReturnContainers returnContainers)
         {
+            if (tracker != null)
+                tracker.Visit(directory.FullName);
```

Comment:
`tracker = new InodeTracker();` - It's not expensive. We can always create this on top level and not check many times during recursion.